provider/provider.go

```diff
@@ -0,0 +1,135 @@
+package provider
+
+import (
+	"github.com/scorum/event-provider-go/event"
+	"github.com/scorum/scorum-go"
+	"github.com/scorum/scorum-go/apis/blockchain_history"
+	"github.com/scorum/scorum-go/apis/chain"
+	"github.com/scorum/scorum-go/transport/http"
+	"gitlab.scorum.com/blog/api/common"
+	"time"
+)
+
+type Config struct {
+	SyncInterval time.Duration
+
+	BlocksHistoryMaxLimit  uint32
+	LookupAccountsMaxLimit uint32
+
+	ErrorRetryTimeout time.Duration
+	ErrorRetryLimit   int
+}
+
+var DefaultConfig = Config{
+	SyncInterval: time.Second,
+
+	BlocksHistoryMaxLimit:  100,
+	LookupAccountsMaxLimit: 1000,
+
+	ErrorRetryTimeout: 10 * time.Second,
+	ErrorRetryLimit:   3,
+}
+
+type Provider struct {
+	Blockchain *scorumgo.Client
+	Config     Config
+}
+
+func NewProvider(url string, config *Config) *Provider {
```

Comment:
passing nil to use the default config looks a bit strange and unexpected. Please consider an Option patter. See e.g. https://github.com/tmrts/go-patterns/blob/master/idiom/functional-options.md